builder/builder.go

```diff
@@ -739,6 +739,55 @@ func (b *Builder) AddBundles(bundles []string, allLocal bool, allUpstream bool,
 	return nil
 }
 
+// RemoveBundles removes a list of bundles from the Mix Bundles List. If a
+// bundle is not present, it is skipped. If 'local' is passed, the corresponding
+// bundle file is removed from local-bundles. Note that this is an irreversible
+// step. The Mix Bundles List is validated when read in, and the resulting Mix
+// Bundles List will be in sorted order.
+func (b *Builder) RemoveBundles(bundles []string, local bool, git bool) error {
+	// Fetch upstream bundle files if needed
+	if err := b.getUpstreamBundles(b.UpstreamVer, true); err != nil {
+		return err
+	}
+
+	// Read in current mix bundles list
+	set, err := b.getMixBundlesListAsSet()
+	if err != nil {
+		return err
+	}
+
+	// Remove the ones passed in from the set
+	for _, bundle := range bundles {
+		fmt.Printf("Removing bundle %s\n", bundle)
+		delete(set, bundle)
+
+		if local {
+			if err := os.RemoveAll(filepath.Join(b.LocalBundleDir, bundle)); err != nil {
```

Comment:
`RemoveAll`? Do we expect this to be a directory?

Reply:
Nope. As @cmarcelo pointed out to me in the past, `os.RemoveAll` doesn't return an error in the event that the file does not exist, and thus you can save yourself the call to `os.Stat` you'd need with `os.Remove` if you're not sure it's there. This kind of short-hand could be dangerous if you're removing files in a shared or more-complicated directory (where you might care, for example, if you're accidentally deleting a directory rather than a file), but this seemed pretty safe to me here.

Reply:
Sure, but `os.RemoveAll` does an `Lstat` of its own (https://golang.org/src/os/path.go?s=1634:1667#L56). This is fine it just seems strange to me to be potentially removing a directory when you only want to remove a file.

Reply:
Yeah -- I meant it saves _our code_ from doing an `Stat`. Just cleaner/less nesting. And I mean, `os.Remove` can remove a directory too. Are you suggesting I have a separate `os.State` to check the `FileInfo` for `IsDir()`?

I do like the idea of supporting nested directories in `local-bundles` down the line (so that you could organize your custom bundles how you pleased) -- almost included that in the original implementation, but figured I'd mirror how `upstream-bundles` worked first, and add that as an optimization if people wanted it. If we start supporting a non-flat `local-bundles`, this section (and a couple other places) would definitely need to be more sophisticated to accommodate that.

Reply:
Ok, fair enough.

Reply:
I decided to change this after all, and include the additional `os.Stat` with additional output, 'cause in testing I found it way easier to understand what's going on.